src/query/execute_schema.c

```diff
@@ -2623,11 +2623,20 @@ do_rename (PARSER_CONTEXT * parser, PT_NODE * statement)
       const char *old_name = current_rename->info.rename.old_name->info.name.original;
       const char *new_name = current_rename->info.rename.new_name->info.name.original;
 
+      /* We cannot change the schema of a class by using synonym names. */
       if (db_find_synonym (old_name))
 	{
 	  PT_ERRORmf (parser, statement, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_CLASS_DOES_NOT_EXIST, old_name);
 	  goto error_exit;
 	}
+      else
+	{
+	  /* db_find_synonym () == NULL */
+	  if (er_errid () == ER_SYNONYM_NOT_EXIST)
+	    {
+	      er_clear ();
+	    }
```

Comment:
db_find_synonym() returned NULL.
If it is due to an error other than ER_SYNONYM_NOT_EXIST, shouldn't it be handled as an error without ignoring it?

 ```
     if (er_errid () == ER_SYNONYM_NOT_EXIST)
	    {
	      er_clear ();
	    }
     else
            {
                // PT_ERRORmf( ... )
                goto error_exit;
            }    
```

Reply:
Added error handling for cases other than ER_SYNONYM_NOT_EXIST errors.